monai/transforms/spatial/array.py

```diff
@@ -489,6 +489,11 @@ def __call__(
         else:
             if self.axcodes is None:
                 raise ValueError("Incompatible values: axcodes=None and as_closest_canonical=True.")
+            if sr < len(self.axcodes):
+                warnings.warn(
+                    f"axcodes ('{self.axcodes}') length is smaller than the number of input spatial dimensions D={sr}.\n"
+                    f"{self.__class__.__name__}: input spatial shape is {spatial_shape}, num. channels is {data_array.shape[0]}."
```

Comment:
I think it's better to add some tips about checking whether the data is already in `channel-first` shape.

Thanks.